lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -2255,10 +2255,13 @@ def update_service(self, name, old_service, new_service):
             name=name,
             **service_data
         )
+        # Unfortunately, docker-py f***ed up and doesn't return the structure
+        # the corresponding API call returns, which would include a list of warnings.
```

Comment:
In what way have they messed up? AFAIK they do not touch the returned json from the docker API and it should contain the `Warnings` key? 

https://docs.docker.com/engine/api/v1.39/#operation/ServiceCreate

Reply:
Ah I see now that they return a boolean.

Reply:
Yep. They simply throw the other information away, which is really annoying...

Reply:
I've opened a PR to return the response https://github.com/docker/docker-py/pull/2272. Lets see what they say and if merged we can add it to `update_serivce`.

Reply:
I kind of doubt that they will change this, for sake of backwards compatibility. A `dict` is always `true`ish, so this PR will break existing programs. But let's see...

Reply:
Yes I know but they never return `False` so it'll "only" break for implementations using it as `return_value is True` (which is redundant anyway as the return value is always True). But yeah let's see..